### ISSUE TITLE
Merge of coqui voices broke stuff

### ISSUE THREAD

**Issue by aedocw** [aedocw/epub2tts#126]
Due to lack of complete testing, the merge that made studio voices work also broke a bunch of other stuff. This merge fixes that, and also includes a test script that I will use in the future to validate a few common use cases. It would be nice to add some real tests into CI, but the test runners do not have GPU, so would not be that useful.

**Comment by danielw97** [comment]
Hi again,
If you'd rather have a separate issue for this let me know, although in my testing just now after pulling your most recent commit I'm getting the following error, as xtts is getting sent a bigger text chunk than it can handle I believe.
This is using one of the coqui studio voices, btw.

Error:  ❗ XTTS can only generate text with a maximum of 400 tokens. ... Retrying (0 retries left)                      
 This is a longer paragraph, although using a finetuned model last week with the same book didn't have this problem.
Thanks for all of your work.

**Comment by aedocw** [comment]
Hmm, I might need to fully revert this then. I have not tested with really
long text, so have not run into exceeding the tokens.

I think it's fair to keep it under this issue as the merge of coqui voices
sure did break stuff!

For what it's worth though, it is working for me with current chunk size
with epubs, maybe I need to test with text longer than what I have sent to
it so far.

**Comment by danielw97** [comment]
Other than that everything seems to be working, I wonder is it possible to encorperate the same segmenting code that is used with xtts as I assume the limits are the same (if it isn't already)?
I think this is an outlier with a longer paragraph that xtts handled fine although I can see how it may cause issues.

**Comment by danielw97** [comment]
Also, the same text processing at least in my mind should be used, as I believe this is basically xtts under the hood unless I am incorrect.

**Comment by aedocw** [comment]
I'm going to reopen this one until things are sorted out.

The difference has to do with how I'm calling XTTS between these two ways. The way I use the xtts cloning model is with their inference streaming approach but the docs don't indicate how you can use that with their voices.

**Comment by danielw97** [comment]
Okay, thanks.
Not a rush especially this time of year with the holidays of course, although wanted to let you know the errors I was seeing.
Maybe they've not implemented the streaming approach with their studio voices, although might be worth asking on the Discord as it might just not be documented.

**Comment by aedocw** [comment]
Haha yes I have asked on discord, no answer yet though (and someone else just asked the same question today).

I put a potential fix in the branch "fixes" if you want to try it out when you get a chance.

As far as the holidays, it's OK, this is relaxing and I always sleep better after fixing some bugs :)

Thanks, and happy holidays to you too!

**Comment by danielw97** [comment]
Thanks, I've got some time this evening and will test this now.
It looks as though it's downloading xtts v1 unless the model's being  called differently. 
In my cache directory I now have tts_models--multilingual--multi-dataset--xtts and tts_models--multilingual--multi-dataset--xtts_v2
Unless this is a change the Coqui team have made very recently.
I'm just waiting for the new model to download and will edit this comment with my findings.

**Comment by aedocw** [comment]
Thanks, I appreciate your testing and all your feedback!

You should see this, indicating the right xtts version:
```Engine is TTS, model is tts_models/multilingual/multi-dataset/xtts_v2                                                                         
 > tts_models/multilingual/multi-dataset/xtts_v2 is already downloaded.                                                                       
 > Using model: xtts ```

**Comment by danielw97** [comment]
Yes, that's what I got in the end.
I made a silly mistake the first time specifying --model instead of --engine although that was fixed fairly quickly, all good now though.

**Comment by aedocw** [comment]
Excellent! I'll figure out what's going on with other languages hopefully tonight and merge this branch.

**Comment by aedocw** [comment]
Found the problem with Coqui voices, reading plain text (rather than epub), and specifying a language other than english. On line 164 I replace all periods with commas if language != en, and that seems to break something along the way (maybe it confuses the segmenter that breaks everything up into individual sentences).

Replacing periods with commas did seem to help for non-english languages where it would seem to always pronounce the period at the end of sentences as "dot" or some variation of that. I changed that to happen now just before the sentence is sent to TTS, hopefully it is still effective for other languages.
